administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
-          else
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_frames,$dest_qual,$dest_imgtype);
+        }
+        
+        if ($metadata)
+        // copy metadata if needed
+        {
+          $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+          if (!$meta_success)
           {
-            JoomFile::fastImageCopyResampled( $dst_img, $src_img, 0, 0, 0, 0, $destWidth,
-                                              (int)$destHeight, $srcWidth, $srcHeight);
-          }
+            $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+            return false;
+          }      
         }
-
-        if(!@imagejpeg($dst_img, $dest_file, $dest_qual))
+        if(!$success)
         {
           // Workaround for servers with wwwrun problem
           $dir = dirname($dest_file);
           JoomFile::chmod($dir, '0777', true);
-          imagejpeg($dst_img, $dest_file, $dest_qual);
+          if ($anim && $imginfo['animation'] && $imginfo['type'] == 'GIF')
+          {
+            JLoader::register('GifCreator', JPATH_COMPONENT_ADMINISTRATOR . '/helpers/GifCreator.php');
+            $gc = new GifCreator();
+            $gc->create($dst_frames, $gfe->getFrameDurations(), 0);
+            $success = file_put_contents($dest_file, $gc->getGif());
+          }
+          else
+          {
+            $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_frames,$dest_qual,$dest_imgtype);
+          }
+          if ($metadata)
+          {
+            $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+            if (!$meta_success)
+            {
+              $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+              return false;
+            }
+          }
           JoomFile::chmod($dir, '0755', true);
         }
-        imagedestroy($src_img);
-        imagedestroy($dst_img);
+        foreach ($src_frames as $key => $frame)
+        {
+          imagedestroy($src_frames[$key]['image']);
+          imagedestroy($dst_frames[$key]);
+        }
         break;
       case 'im':
+        $debugoutput.='ImageMagick...<br/>';
         $disabled_functions = explode(',', ini_get('disabled_functions'));
         foreach($disabled_functions as $disabled_function)
         {
           if(trim($disabled_function) == 'exec')
           {
             $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_OUTPUT_EXEC_DISABLED').'<br />';
-
             return false;
           }
         }
-
-        if(!empty($config->jg_impath))
+        @exec(trim($config->jg_impath).'convert -version', $output_convert);
+        @exec(trim($config->jg_impath).'magick -version', $output_magick);
+        if ($output_convert)
         {
-          $convert_path=$config->jg_impath.'convert';
+          $convert_path=trim($config->jg_impath).'convert';
+        }
+        elseif ($output_magick)
```

Comment:
If ImageMagick v6.x is not available but v7.x, use "magick convert" for resizing